### PR TITLE
EAHW-2172: Create UI ajax call for getting all time entries

### PR DESCRIPTION
**Pull request by daleelR**
add get time entries ajax call and test. add mock data for get time entries.

### CODE REVIEW

**Comment by LaurieBennett** [comment]
Do we not need to do any conversion for the data into the format used in the UI?

**Comment by daleelR** [comment]
> Do we not need to do any conversion for the data into the format used in the UI?

**Comment by laurieboyes** [comment]
Nice thanks for doing this 👍 Do you feel that this would be a good time to add a mock for the endpoint to json server?

**Comment by daleelR** [comment]
> Nice thanks for doing this 👍 Do you feel that this would be a good time to add a mock for the endpoint to json server?

So there's a mock endpoint in `mocks/apiServer.js` there for get already, were you referring to something else?

**Comment by daleelR** [comment]
> Do we not need to do any conversion for the data into the format used in the UI?

@LaurieBennett Have now added

File: src/utils/time-entry-utils/timeEntryUtils.js [hunk 1]
```diff
@@ -0,0 +1,15 @@
+import dayjs from 'dayjs';
+
+export const getSingleTimeEntryResponseItem = (timeEntryResponseData) => {
+  if (timeEntryResponseData.items && timeEntryResponseData.items.length > 0)
+    return timeEntryResponseData.items[0];
+  else return null;
```
Comment (LaurieBennett) [inline src/utils/time-entry-utils/timeEntryUtils.js:6]: Should we throw an error here as well to inform a developer what the issue is?

**Review by LaurieBennett — CHANGES_REQUESTED** [review]

File: src/utils/time-entry-utils/timeEntryUtils.js (hunk 1 above)
Comment (daleelR) [reply src/utils/time-entry-utils/timeEntryUtils.js:6]: Ah yeah I was contemplating this. Think you're right.
I think on the 'else' we should not return anything and just throw an error as then it conforms to the name/purpose of this function.

Comment (daleelR) [reply src/utils/time-entry-utils/timeEntryUtils.js]: done

**Review by LaurieBennett — DISMISSED** [review]
Looks good, PASS

**Review by LaurieBennett — APPROVED** [review]
re-approved

**Review by laurieboyes — APPROVED** [review]

**Pull request merged**